addon/create-class-computed.js

```diff
@@ -0,0 +1,93 @@
+import Ember from 'ember';
+import EmberObject from 'ember-object';
+import computed from 'ember-computed';
+import on from 'ember-evented/on';
+import observer from 'ember-metal/observer';
+import { isNone } from 'ember-utils';
+// import getOwner from 'ember-owner/get';
+import get from 'ember-metal/get';
+import { setProperties } from 'ember-metal/set';
+import WeakMap from 'ember-weakmap';
+import { mapKeysToValues } from './-build-computed';
+import getValue from './get-value';
+import collapseKeys from './collapse-keys';
+import flattenKeys from './flatten-keys';
+
+const { defineProperty } = Ember;
+
+const PROPERTIES = new WeakMap();
+
+function findOrCreatePropertyInstance(context, propertyClass, key) {
+  let propertiesForContext = PROPERTIES.get(context);
+  if (isNone(propertiesForContext)) {
+    propertiesForContext = {};
+    PROPERTIES.set(context, propertiesForContext);
+  }
+
+  let property = propertiesForContext[key];
+  if (property) {
+    return property;
+  }
+
+  // let owner = getOwner(context);
+  property = propertyClass.create(/*owner.ownerInjection(), */{
+    _key: key,
+    _context: context
+  });
+
+  propertiesForContext[key] = property;
+  return property;
+}
+
+const BaseClass = EmberObject.extend({
+  _computedDidChange: observer('_computed', function() {
+    this._context.notifyPropertyChange(this._key);
+  })
+});
+
+function createObserverClass(classKeysObj, macroGenerator) {
+  let classKeys = Object.keys(classKeysObj);
+  let observers = classKeys.filter(key => classKeysObj[key]);
+
+  function rewriteComputed() {
+    let values = mapKeysToValues(observers, getValue, this);
+    let cp = macroGenerator(...values);
+    defineProperty(this, '_computed', cp);
+  }
+
+  let classProperties = observers.reduce((properties, key) => {
```

Comment:
The caller tells us which keys to observe for changes [here](https://github.com/kellyselden/ember-awesome-macros/blob/create-class-computed/addon/array/filter-by.js#L7-L11).
Then we filter out the non-observers [here](https://github.com/kellyselden/ember-macro-helpers/blob/create-class-computed/addon/create-class-computed.js#L50).